email/cache.go

```diff
@@ -0,0 +1,68 @@
+package email
+
+import (
+	"crypto/sha256"
+	"encoding/hex"
+	"sync"
+
+	"github.com/golang/groupcache/lru"
+	"github.com/prometheus/client_golang/prometheus"
+)
+
+type EmailCache struct {
+	sync.Mutex
+	cache    *lru.Cache
+	requests *prometheus.CounterVec
+}
+
+func NewHashedEmailCache(maxEntries int, stats prometheus.Registerer) *EmailCache {
+	requests := prometheus.NewCounterVec(prometheus.CounterOpts{
```

Comment:
This metric is going to be skewed: most cache hits will happen at ingestion time (exporter.go:110), thereby immediately stopping the processing of the duplicate email.

Most cache misses will happen _twice_, both at ingestion time and again at processing time (pardot.go:148).

That means that this metric won't be measuring what we think it is measuring, and may lead to erroneous conclusions later on.